lib/charms/prometheus_k8s/v0/prometheus_remote_write.py

```diff
@@ -222,47 +234,47 @@ def _from_file(self, root_path: Path, file_path: Path) -> List[dict]:
 
             except Exception as e:
-                logger.error("Failed to read alert rules from %s: %s", file_path.name, e)
+                logger.error("Failed to read rules from %s: %s", file_path.name, e)
                 return []
 
-            if _is_official_alert_rule_format(rule_file):
-                alert_groups = rule_file["groups"]
-            elif _is_single_alert_rule_format(rule_file):
-                # convert to list of alert groups
+            if _is_official_rule_format(rule_file):
+                groups = rule_file["groups"]
+            elif _is_single_rule_format(rule_file, self.rule_type):
+                # convert to list of groups
                 # group name is made up from the file name
-                alert_groups = [{"name": file_path.stem, "rules": [rule_file]}]
+                groups = [{"name": file_path.stem, "rules": [rule_file]}]
             else:
                 # invalid/unsupported
                 logger.error("Invalid rules file: %s", file_path.name)
                 return []
 
             # update rules with additional metadata
-            for alert_group in alert_groups:
-                if not self._is_already_modified(alert_group["name"]):
+            for group in groups:
+                if not self._is_already_modified(group["name"]):
                     # update group name with topology and sub-path
-                    alert_group["name"] = self._group_name(
+                    group["name"] = self._group_name(
                         str(root_path),
                         str(file_path),
-                        alert_group["name"],
+                        group["name"],
                     )
 
                 # add "juju_" topology labels
-                for alert_rule in alert_group["rules"]:
-                    if "labels" not in alert_rule:
-                        alert_rule["labels"] = {}
+                for rule in group["rules"]:
+                    if "labels" not in rule:
+                        rule["labels"] = {}
 
                     if self.topology:
                         # only insert labels that do not already exist
                         for label, val in self.topology.label_matcher_dict.items():
-                            if label not in alert_rule["labels"]:
-                                alert_rule["labels"][label] = val
-                        # insert juju topology filters into a prometheus alert rule
-                        alert_rule["expr"] = self.tool.inject_label_matchers(
-                            re.sub(r"%%juju_topology%%,?", "", alert_rule["expr"]),
+                            if label not in rule["labels"]:
+                                rule["labels"][label] = val
+                        # insert juju topology filters into a prometheus rule
+                        rule["expr"] = self.tool.inject_label_matchers(
+                            re.sub(r"%%juju_topology%%,?", "", rule["expr"]),
```

Comment:
Should we emit a deprecation warning if we encounter the %% stub?

Reply:
Not if there's any hope of extracting this into a "unified" set of operations via this kind of abstraction, rather than scrape, remote-write, loki, and everything else having their own 98% compatible implementations.

Loki needs this (not replaced with `""`, but there's nothing we can deprecate it for at the moment)